src/client/menu/index.js

```diff
@@ -73,14 +74,37 @@ export default class Menu extends React.Component {
     this.setState({ users })
     await this.handleGitUserChanges()
   }
-  handleAddUser = async newUser => {
-    const users = userService.add(newUser)
-    this.setState({ users })
+  handleAddUser = () => {
+    this.setState({
+      userEdits: { name: '', email: '', rsaKeyPath: '', active: true }
+    })
+  }
+  handleSubmitAddUser = async () => {
+    const users = userService.add(this.state.userEdits)
+    this.setState({
+      users,
+      userEdits: null
+    })
     await this.handleGitUserChanges()
   }
-  handleEditUser = async user => {
-    const users = userService.update(user)
-    this.setState({ users })
+  handleUserFormSubmit = () => {
+    this.state.userEdits.id ? this.handleSubmitEditUser() : this.handleSubmitAddUser()
+  }
+  handleEditUser = user => {
+    this.setState({ userEdits: { ...user } })
+  }
+  handleUserEditsChange = user => {
+    this.setState({ userEdits: user })
+  }
+  handleCancelEditUser = () => {
```

Comment:
Looks like this is the cancel for both add & edit, so maybe just `handleCancel`?